rlp/codec.py

```diff
@@ -120,29 +105,29 @@ def length_prefix(length, offset):
                    list
     """
     if length < 56:
-        return ascii_chr(offset + length)
+        return struct.pack('B', offset + length)
     elif length < 256**8:
         length_string = int_to_big_endian(length)
-        return ascii_chr(offset + 56 - 1 + len(length_string)) + length_string
+        return struct.pack('B', offset + 56 - 1 + len(length_string)) + length_string
```

Comment:
I think `chr(intval)` here is a little clearer than `struct.pack('B', intval)`, since it's exactly one int. It also has a 30% performance benefit:

```py
In [145]: def profile(to_wrap, runs=10000):
     ...:     start = time.time()
     ...:     for _ in range(runs): to_wrap()
     ...:     duration = time.time() - start
     ...:     print(f'took {duration} s to run {runs} times. {duration*1000000/runs} us per run')

In [148]: profile(lambda : struct.pack('B', 157), runs=10000000)
took 2.092618227005005 s to run 10000000 times. 0.20926182270050048 us per run

In [149]: profile(lambda : chr(157), runs=10000000)
took 1.4498624801635742 s to run 10000000 times. 0.14498624801635743 us per run
```

Reply:
`chr` returns a string so I can't use it on it's own.

```
struct.pack
took 0.18044090270996094 s to run 100000 times. 1.8044090270996094 us per run
chr(v)
took 0.16496515274047852 s to run 100000 times. 1.6496515274047852 us per run
chr(v).encode("latin1")
took 0.18259096145629883 s to run 100000 times. 1.8259096145629883 us per run
bytes([v])
took 0.1914830207824707 s to run 100000 times. 1.914830207824707 us per run
```

I'm not seeing anything faster than `struct.pack`.

Reply:
:man_facepalming: of course

Reply:
But https://github.com/ethereum/pyrlp/pull/63 is faster.  Hurray for really dumb solutions!